Collapsible resizers

tofarr opened the pull request:
Support for collapsible panels in the UI
<img width="1642" alt="image" src="https://github.com/user-attachments/assets/d142b414-2b04-4247-b1cd-ffacf51d3a7a">

Separators are still draggable, but users can now click on the Icons to expand / collapse panels:
<img width="1608" alt="image" src="https://github.com/user-attachments/assets/f87c64a8-e591-459c-a7c0-8dbdbaa6514d">

<img width="1610" alt="image" src="https://github.com/user-attachments/assets/d9388b90-5170-4b84-9e0a-1f00a4ab7b85">

<img width="1604" alt="image" src="https://github.com/user-attachments/assets/e6b7caa8-5a7b-4810-b6dd-33d54ba72ce5">

neubig commented:
@amanape , could you take a look? If you're busy I'll grab it though.

amanape commented on frontend/src/components/Resizable.tsx:
```diff
@@ -67,33 +90,104 @@ export function Container({
-        `flex ${orientation === Orientation.HORIZONTAL ? "" : "flex-col"}`,
-        className,
-      )}
+      className={twMerge(`flex ${isHorizontal ? "" : "flex-col"}`, className)}
```
```suggestion
      className={twMerge("flex", isHorizontal && "flex-col", className)}
```

amanape reviewed (approved):
LGTM! Though you may consider wrapping some of the handlers in a `useCallback` since this component is re-render-intensive

amanape commented:
@tofarr Does this resolve #3199 / #1745?

tofarr commented:
> @tofarr Does this resolve #3199 / #1745?

I think so - panels are now resizable and no longer have a min size. Though on of the changes since deploy seems to have done some thing odd

tofarr commented:
The fix to isHorizontal inadvertently inverted the logic, breaking the layout - so I added one more fix:
Broken:
![image](https://github.com/user-attachments/assets/70adc9e4-b294-4fb0-ba86-fa7c2e1baad4)

Fixed:
![image](https://github.com/user-attachments/assets/d34691ff-7a5f-42e5-ae6c-e29043d1c289)
